js/widgets/popup.js

```diff
@@ -865,14 +857,16 @@ $.widget( "mobile.popup", {
 			// If that happens and we remove the container a few lines below, we
 			// will cause an infinite recursion - #5244
 			.detach()
-			.insertAfter( this._ui.placeholder )
-			.removeClass( "ui-popup ui-overlay-shadow ui-corner-all ui-body-inherit" );
+			.insertAfter( this._ui.placeholder );
 		this._ui.screen.remove();
 		this._ui.container.remove();
 		this._ui.placeholder.remove();
 	},
 
 	_destroy: function() {
+		if ( this.options.enhanced ) {
+			this.classesElementLookup = {};
+		}
```

Comment:
this seems very questionable? this complete prevents auto destroy from working? Including classes automatically added by the widget factory.